live-hls.js

```diff
@@ -606,16 +675,68 @@ master = function(request, response) {
  */
 
 event = function(request, response) {
+
   fs.readFile(path.join(__dirname, 'data', request.path), function(error, data) {
-    var event, playlist, result;
+    var event, playlist, result, tsstreampath, stream;
     if (error) {
       return response.send(404, error);
     }
 
-    event = extend(getStream('event/' + request.path), request.params);
-
+    event = extend(getStream('event' + request.path), request.params);
     playlist = data.toString();
-    result = filterPlaylist(playlist,
+    if (manifest['event' + request.path] == undefined) {
+
+      playlist = data.toString();
+      manifestHeader = getHeaderObjects(playlist);
+      manifestResources = getResources(playlist, request);
+      manifest['event' + request.path] =
+      {
+        header: manifestHeader,
+        resources: manifestResources
+      };
+    }
+    if (event.resetStream==1) {
+      event.resetStream = 0;
+      resetLiveStream(streampath);
+    }
+    if (event.resetStream==2) {
+      event.resetStream = 0;
+      resetAllStreams();
+    }
+
+    if (event.stopStream==1) {
+      event.resetStream = 0;
+      stopLiveStream(streampath);
+    }
+
+    if (event.stopStream==2) {
+      event.resetStream = 0;
+      stopAllStreams();
+    }
+
+    if (event.tsnotfound>0) {
+      //Pick a future .ts file to inject 404
+      tsstreampath = manifest['event' + request.path].resources[event.lastStartPosition + event.window + 1].tsfile;
+      tsstreampath = trimCharacters(tsstreampath, ['/','.']);
+      console.log('Target for 404: ' + tsstreampath);
+      stream = getStream(tsstreampath);
+      //Pass error value to the ts stream
+      if (stream.tsnotfound) {
+        stream.tsnotfound++;
+      }
+      else {
+        stream.tsnotfound = 1;
+      }
+      event.tsnotfound--;
+    }
+    if (event.manifestnotfound>0) {
+      if (processErrors(request, response, event) == true) {
+        console.log('errors processed manifestnotfound=' + event.manifestnotfound);
+        return response;
+      }
+    }
+    event.rate = event.rate ? event.rate : manifestHeader.TargetDuration.value;
+    result = filterPlaylist( manifest['event' + request.path],playlist,
```

Comment:
`filterPlaylist()` and `extractResourceWindow()` also share a lot of the same logic and just differ on a few key calculations like start and end positions. We should make use of `extractResourceWindow()` in the consolidated function I mentioned in my other comment about `event()` and `live()` and make sure the inputs are set correctly to get the desired output we want (event or live style)